core/partitioning/SegmentedBlock.h

```diff
@@ -0,0 +1,89 @@
+#pragma once
+
+#include <vector>
+
+#include "NvInfer.h"
+#include "core/ir/ir.h"
+#include "core/partitioning/PartitionInfo.h"
+#include "torch/csrc/jit/ir/ir.h"
+
+namespace trtorch {
+namespace core {
+namespace partitioning {
+
+struct SegmentedBlock {
+ public:
+  enum SegmentedBlockTarget {
+    kTorch,
+    kTensorRT,
+  };
+
+  SegmentedBlock() = default;
+  SegmentedBlock(SegmentedBlockTarget blk_target) : target_(blk_target), g_(std::make_shared<torch::jit::Graph>()) {}
+  SegmentedBlock(SegmentedBlockTarget blk_target, std::vector<torch::jit::Node*>& nodes);
+  SegmentedBlock(SegmentedBlockTarget blk_target, std::shared_ptr<torch::jit::Graph> g) : target_(blk_target), g_(g) {}
+
+  torch::jit::Value* getOrAddInputForValue(torch::jit::Value* v);
+  torch::jit::Node* cloneNode(torch::jit::Node* node);
+  void appendNode(torch::jit::Node* n) {
+    cloneNode(n);
+  }
+  void registerOutput(torch::jit::Value* raw_output);
+  torch::jit::graph_node_list nodes() {
+    return g_->nodes();
+  }
+  const std::vector<torch::jit::Node*>& raw_nodes() const {
+    return nodes_;
+  }
+  torch::jit::Block* block() {
+    return g_->block();
+  }
+  std::shared_ptr<torch::jit::Graph>& g() {
+    return g_;
+  }
+  void update_graph(std::shared_ptr<torch::jit::Graph> new_g) {
+    g_ = new_g;
+  }
+  c10::ArrayRef<torch::jit::Value*> inputs() {
+    return g_->inputs();
+  }
+  c10::ArrayRef<torch::jit::Value*> outputs() {
+    return g_->outputs();
+  }
+  const std::vector<torch::jit::Value*>& raw_inputs() const {
+    return inputs_;
+  }
+  const std::vector<torch::jit::Value*>& raw_outputs() const {
+    return outputs_;
+  }
+  void eraseInput(size_t i);
+  void eraseOutput(size_t i);
+  bool contain_raw_value(torch::jit::Value* input) {

```

Comment:
what is a raw value?

Reply:
Raw value is the torch::jit::Value* from original lowered graph, we use this function to identify if current block produces any torch::jit::Value* that later block will use as inputs.